ports/atmel-samd/tick.c

```diff
@@ -52,21 +52,34 @@ void SysTick_Handler(void) {
 
 void tick_init() {
     uint32_t ticks_per_ms = common_hal_mcu_processor_get_frequency() / 1000;
-    SysTick_Config(ticks_per_ms);
+    SysTick_Config(ticks_per_ms-1);
     NVIC_EnableIRQ(SysTick_IRQn);
 }
 
 void tick_delay(uint32_t us) {
     uint32_t ticks_per_us = common_hal_mcu_processor_get_frequency() / 1000 / 1000;
-    uint32_t us_between_ticks = SysTick->VAL / ticks_per_us;
-    uint64_t start_ms = ticks_ms;
+    uint32_t us_until_next_tick = SysTick->VAL / ticks_per_us;
+    uint32_t start_tick;
     while (us > 1000) {
-        while (ticks_ms == start_ms) {}
-        us -= us_between_ticks;
-        start_ms = ticks_ms;
-        us_between_ticks = 1000;
+        // check if interrupts are disabled
+        if(__get_PRIMASK())
+           return; // if not just return
```

Comment:
We don't need this if we don't use ticks_ms right? SysTick->VAL will still be updated.

Reply:
I thought you wanted to exit if the caller "forgot" to disable interrupt for a long delay. That is what this if for. If callers behave and disable interrupts only for us<100 then this is not needed.
If we fix pulseio as in this PR then it is "well behaved but I don't know about other callers of tick_delay.  The bad thing about this is it gives a very short delay "silently". It will likely break something.

Reply:
We don't need this because it doesn't depend on `ticks_ms` being updated by the interrupt. I wanted earlier to prevent the lock up situation where we were waiting for `ticks_ms` to change but the interrupt was disabled.